app/validators/placeholders/placeholder_validator.py

```diff
@@ -143,24 +143,32 @@ def validate_answer_and_transform_unit_match(self, *, arguments, transform_type)
         value = arguments["value"]
         answer_id = value.get("identifier")
         unit = arguments["unit"]
+        source_answer_ids = []
 
-        if self.errors or (
-            unit
-            == self.questionnaire_schema.answers_with_context[answer_id]["answer"][
-                "unit"
-            ]
-        ):
-            return None
+        if value.get("source") == "calculated_summary":
+            if calculated_summary_source_answer_ids := self.questionnaire_schema.get_calculation_block_ids(
+                block=self.questionnaire_schema.get_block(value.get("identifier")),
```

Comment:
if you go with `identifier = value.get("identifier")` above, you can reuse that here